### PR TITLE
Apply deprecation decorator.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation
Followup #1382. 

## Description of the changes

This PR applies the deprecation decorator to the following classes:

- `optuna.integratoin.CmaEsSampler`
- `optuna.struct.FrozenTrial`
- `optuna.struct.StudySummary`
- `optuna.struct.TrialPruned`

### CODE REVIEW

File: optuna/structs.py [hunk 1]
```diff
@@ -314,21 +314,15 @@ def __le__(self, other):
         return self._study_id <= other._study_id
 
 
+@deprecated(
+    "0.19.0",
+    "3.0.0",
+    text=(
+        "This class was moved to :mod:`~optuna.exceptions`. Please use"
```
Comment (hvy) [inline optuna/structs.py:321]:
```suggestion
        "This class was moved to :mod:`~optuna.exceptions`. Please use "
```

File: optuna/structs.py [hunk 2]
```diff
@@ -231,14 +228,17 @@ def duration(self):
             return None
 
 
+@deprecated(
+    "1.4.0",
+    "3.0.0",
+    text=(
+        "This class was moved to :mod:`~optuna.study`. Please use"
```
Comment (hvy) [inline optuna/structs.py:235]:
```suggestion
        "This class was moved to :mod:`~optuna.study`. Please use "
```

File: tests/test_structs.py [hunk 3]
```diff
@@ -42,3 +41,9 @@ def test_study_summary_deprecated():
             datetime_start=datetime.datetime.now(),
             study_id=0,
         )
+
+
+def test_trial_pruned_deprecated() -> None:
```
Comment (hvy) [inline tests/test_structs.py:46]: 👍

**Comment by HideakiImamura** [comment]
@toshihikoyanase  I have another comment. How about applying the deprecation decorator to `optuna.integration.lightgbm.LightGBMTuner.best_booster`?

**Comment by toshihikoyanase** [comment]
@hvy @HideakiImamura Thank you for your comment on the `optuna.integration.lightgbm.LightGBMTuner.best_booster` property. I applied the decorator in https://github.com/optuna/optuna/pull/1413/commits/a3988b4234b29ee1537f006a1b75ff7b72796a5a, and confirmed that the decorator worked as expected. PTAL.

image

**Comment by toshihikoyanase** [comment]
Mypy does not support the type checking of decorated properties.

> optuna/integration/_lightgbm_tuner/optimize.py:732: error: Decorated property not supported

As suggested in https://github.com/python/mypy/issues/1362, I add `# type: ignore` to the `@propety`.

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 4]
```diff
@@ -728,18 +729,17 @@ def __init__(
...
-    @property
+    @property  # type: ignore
+    @deprecated(
+        "1.4.0",
+        "3.0.0",
+        text=(
+            "Please get the best booster via "
+            ":class:`~optuna.integration.lightgbm.LightGBMTuner.get_best_booster` instead."
+        ),
+    )
```
Comment (hvy) [inline optuna/integration/_lightgbm_tuner/optimize.py:740]: Mostly wondering but does having the `deprecated` also work?

Comment (toshihikoyanase) [reply optuna/integration/_lightgbm_tuner/optimize.py:740]: I'm sorry but I couldn't get the point. If you mean the confirmation of the behavior of `best_booster` property with `@decorated`, it works as expected :

```python
# examples/lightgbm_tuner_simple.py

    ....
    # model = lgb.train(
    #     params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
    # )
    # model = lgb.train(
    #     params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
     # )
    lgbt = lgb.LightGBMTuner(
        params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
    )
    lgbt.run()
    model = lgbt.best_booster
```

```console
optuna/_deprecated.py:110: DeprecationWarning: best_booster has been deprecated in v1.4.0. This feature will be removed in v3.0.0.
  DeprecationWarning,
```

Comment (hvy) [reply optuna/integration/_lightgbm_tuner/optimize.py:740]: Sorry, I meant having the `deprecated` decorator outermost (applied after `property`).

**Pull request merged**